apps/oc_chef_authz/src/oc_chef_policy_group.erl

```diff
@@ -0,0 +1,180 @@
+%% -*- erlang-indent-level: 4;indent-tabs-mode: nil; fill-column: 92 -*-
+%% ex: ts=4 sw=4 et
+%% @author Daniel DeLeo <dan@chef.io>
+%% Copyright 2012-2015 Opscode, Inc. All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+
+-module(oc_chef_policy_group).
+
+-include("../../include/oc_chef_types.hrl").
+-include_lib("mixer/include/mixer.hrl").
+
+-behaviour(chef_object).
+
+-mixin([{chef_object,[{default_fetch/2, fetch}]}]).
+
+-export([
+         parse_binary_json/1,
+         flatten/1,
+         delete/2,
+         create_record/3
+        ]).
+
+%% chef_object behaviour callbacks
+-export([
+         id/1,
+         authz_id/1,
+         bulk_get_query/0,
+         create_query/0,
+         delete_query/0,
+         ejson_for_indexing/2,
+         fields_for_fetch/1,
+         fields_for_update/1,
+         find_query/0,
+         is_indexed/0,
+         list/2,
+         list_query/0,
+         name/1,
+         new_record/3,
+         org_id/1,
+         record_fields/0,
+         set_created/2,
+         set_updated/2,
+         type_name/1,
+         update_from_ejson/2,
+         update_query/0,
+         update/2
+        ]).
+
+-define(VALIDATION_CONSTRAINTS,
+        {[{<<"name">>, {string_match, chef_regex:regex_for(policy_file_name)}}]}).
+
+id(#oc_chef_policy_group{id = Id}) ->
+    Id.
+
+name(#oc_chef_policy_group{name = Name}) ->
+    Name.
+
+org_id(#oc_chef_policy_group{org_id = OrgId}) ->
+    OrgId.
+
+type_name(#oc_chef_policy_group{}) ->
+    policy.
+
+authz_id(#oc_chef_policy_group{authz_id = AuthzId}) ->
+    AuthzId.
+
+
+create_query() ->
+    insert_policy_group.
+
+update_query() ->
+    update_policy_group_by_id.
+
+delete_query() ->
+    delete_policy_group_by_id.
+
+find_query() ->
+    find_policy_group_by_orgid_name.
+
+list_query() ->
+    list_policy_groups_for_org.
+
+bulk_get_query() ->
+    %% TODO: do we need this?
+    ok.
+
+new_record(OrgId, AuthzId, PolicyGroupData) ->
+    Name = ej:get({<<"name">>}, PolicyGroupData),
+    Id = chef_object_base:make_org_prefix_id(OrgId, Name),
+    #oc_chef_policy_group{
+        id = Id,
+        authz_id = AuthzId,
+        org_id = OrgId,
+        name = Name}.
+
+create_record(OrgId, Name, RequestingActorId) ->
+    Policy = #oc_chef_policy_group{
+                           org_id = OrgId,
+                           name = Name},
+    set_created(Policy, RequestingActorId).
```

Comment:
We really need to make this a default behavior - every module has this extra step, no reason  not to have the framework do it. (Not here and now, I just wanted to get this written somewhere so I don't forget it...)